src/components/panel/panel.js

```diff
@@ -941,13 +999,15 @@ MdPanelRef.prototype.close = function() {
   var self = this;
 
   return this._$q(function(resolve, reject) {
-    var done = self._done(resolve, self);
-    var detach = self._simpleBind(self.detach, self);
-
-    self.hide()
-        .then(detach)
-        .then(done)
-        .catch(reject);
+    self._callInterceptors(MdPanelRef.interceptorTypes.CLOSE).then(function() {
+      var done = self._done(resolve, self);
```

Comment:
`function _done()` is VERY confusing. Similar to the angular core code `valueFn`, use

```js
function _callbackFn() { return function callbackRef() { return callback(self); } }
```